test/models/report_test.rb

```diff
@@ -4,7 +4,39 @@
 require 'test_helper'
 
 class ReportTest < ActiveSupport::TestCase
-  test 'Valid Report' do
+  test 'Valid Report with type hate' do
```

Comment:
I think it would be better to substitute all your 'Valid report with type X' tests with a 'Report with valid type' and check the `type_of` field instead of comparing strings. You can use `assert_includes range, value` to check if the field lies inside the valid range of values.

This way, I think the tests will be simpler and clearer (and also will work with any translation of the app).

Reply:
I've seen that you've uploaded some changes. What I was trying to say is having a single 'Report with valid type' that would substitute all your tests that are currently checking all types (lines 7-25, along with your private method `test_report_type`. 

For that, I would write a test such as this one:

```ruby
test 'Report with valid type' do
  # Setup user and meetup records
  user = users(:one)
  meetup = meetups(:one)

  # List of types of report records
  expected_types = {'0': 'hate', '1': 'menace', '2': 'pornography', '3': 'community', '4': 'other'}

  # Iterate through the list
  expected_types.each do |type, type_name|
    # Create a report with each of the possible types
    report = Report.new(title: 'TestTitle', type_of: type, 
                                       reportable_type: 'Meetup',
                                       user_id: user.id,
                                       reportable_id: meetup.id,
                                       description: 'TestDescritpion')

    # Check if report record is valid
    assert_equal report.valid?, true, 'Report is invalid with all info'
    # Check if the type string is correct.
    assert_equal report.text_type, I18n.t("report.type.#{type_name}")
    # Check if status string is correct
    assert_equal report.text_status, I18n.t('report.status.open')
    # Check if reportable title is correct
    assert_equal report.reportable_title, meetup.title
  end
end
```

This way, with a single test you can verify all possible types and for adding one you will only have to add it to the `expected_types` collection. Also, I've used `I18n.t` for checking the translated version of the values.

I believe this way we can keep our code a bit simpler and we won't repeat ourselves that much. What do you think?

Reply:
In this case I think we can check if the `type_of` is valid or not in a separate unit test, since we are using each unit test for each validation in our model. We check all `presence: true` validations in the model and I believe we should use a single unit test for the `type_of` validation (unit test per validation). 

But there are several approaches apart from this one, so if you think there’s another option we should consider you can give it a try. The main concern about our unit tests is to keep them as simple and clean as we can 😉